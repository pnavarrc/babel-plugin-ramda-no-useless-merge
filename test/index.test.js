import pluginTester from "babel-plugin-tester";
import noUselessRamdaMerge from "../src";

pluginTester({
  plugin: noUselessRamdaMerge,
  babelOptions: {
    retainLines: true
  },
  tests: [
    {
      title: "R.merge with 2 identifiers",
      code: "R.merge(a, b)",
      snapshot: true
    },
    {
      title: "R.merge with 2 objects",
      code: "R.merge({ name: 'a' }, { name: 'b' })",
      snapshot: true
    },
    {
      title: "R.merge with R.merge",
      code: `
      R.merge({ name: 'a' }, R.merge(a, b));
      `,
      snapshot: true
    },
    {
      title: "R.merge with more nested R.merge",
      code: `
      R.merge({ name: 'a' }, R.merge(a, R.merge(b, c)));
      `,
      snapshot: true
    },
    {
      title: "R.merge with function arguments",
      code: `
      const c = R.merge(createObject(), getOptions());
      `,
      snapshot: true
    },
    {
      title: "R.merge not being called",
      code: `
      const Rmerge = R.merge;
      `,
      snapshot: false
    },
    {
      title: "R.merge with one argument",
      code: `
      R.merge({ name: 'a' });
      `,
      snapshot: false
    },
    {
      title: "R.merge with 3 arguments",
      code: `
      R.merge({ name: 'a' }, b, c);
      `,
      snapshot: false
    }
  ]
});
