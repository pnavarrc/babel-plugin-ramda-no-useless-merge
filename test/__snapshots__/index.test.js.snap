// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ramda/no-useless-merge R.merge with 2 identifiers: R.merge with 2 identifiers 1`] = `
"
R.merge(a, b)

      ↓ ↓ ↓ ↓ ↓ ↓

({ ...a, ...b });
"
`;

exports[`ramda/no-useless-merge R.merge with 2 objects: R.merge with 2 objects 1`] = `
"
R.merge({ name: 'a' }, { name: 'b' })

      ↓ ↓ ↓ ↓ ↓ ↓

({ ...{ name: 'a' }, ...{ name: 'b' } });
"
`;

exports[`ramda/no-useless-merge R.merge with R.merge: R.merge with R.merge 1`] = `
"
R.merge({ name: 'a' }, R.merge(a, b));

      ↓ ↓ ↓ ↓ ↓ ↓

({ ...{ name: 'a' }, ...{ ...a, ...b } });
"
`;

exports[`ramda/no-useless-merge R.merge with function arguments: R.merge with function arguments 1`] = `
"
const c = R.merge(createObject(), getOptions());

      ↓ ↓ ↓ ↓ ↓ ↓

const c = { ...createObject(), ...getOptions() };
"
`;

exports[`ramda/no-useless-merge R.merge with more nested R.merge: R.merge with more nested R.merge 1`] = `
"
R.merge({ name: 'a' }, R.merge(a, R.merge(b, c)));

      ↓ ↓ ↓ ↓ ↓ ↓

({ ...{ name: 'a' }, ...{ ...a, ...{ ...b, ...c } } });
"
`;
